### PR TITLE
cleanup & fix bug

### PR DESCRIPTION
**Pull request by izturn**
Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

### CODE REVIEW

**Comment by ross-at-finix** [comment]
Thanks! Can you detail what bugs were found, and which changes correct them?

**Comment by wrossmorrow** [comment]
Thanks! Can you detail what bugs were found, and which changes correct them?

File: server.go [hunk 1]
```diff
@@ -28,16 +27,16 @@ func Serve(port int, processor RequestProcessor) {
...
+		processor: &processor,
+		options:   opts,
 	}
 	epb.RegisterExternalProcessorServer(s, extproc)
 	hpb.RegisterHealthServer(s, &HealthServer{})
 
 	log.Printf("Starting ExtProc(%s) on port %d\n", name, port)
 
-	var gracefulStop = make(chan os.Signal)
+	var gracefulStop = make(chan os.Signal, 1)
```
Comment (izturn) [inline server.go:39]: @wrossmorrow here,  accord to the comments from std package:
>
> // Package signal will not block sending to c: the caller must ensure
> // that c has sufficient ***buffer*** space to keep up with the expected
> // signal rate. For a channel used for notification of just one signal value,
// a buffer of size 1 is sufficient.
`

**Comment by izturn** [comment]
Btw, perhaps the `RequestId` need change to `RequestID`, but that may lead to breaking change, so I don't do that

```go
type RequestContext struct {
    ...
    RequestId   string 
`

**Comment by ross-at-finix** [comment]
> Btw, perhaps the `RequestId` need change to `RequestID`, 

Yeah I'm not really aware of a clean standard here. Some places the initialism is properly capitalized, some it isn't. I can review and see what would really break.

**Pull request merged**